### ISSUE TITLE
Error "argument must be coercible to non-negative integer" when I just change the "intersect" argument input

### ISSUE THREAD

**Issue by sharathkmenon** [krassowski/complex-upset#101]
I want to make complex plot for specific intersections, and I was facing some issues.
I have given a simplified version of the code below in order to show the issue I, where I just want to plot the intersection between two sets. The behavior is different depending on the "intersect" input. The code gives an error (given below) when I give  all the sets as input, but works fine when I just give the two specific sets as input (rather than a vector with all the sets)

Here, I am asking for inclusive intersections, so ideally it should not matter if the 'intersect' argument includes other sets or not.  

NOTE:- I should mention that this is NOT an issue for the ggplotmovies2 dataframe. Thus something must be wrong with the data frame that I am using but I cant really figure it out. 

Case 1:- When I give all the sets (stored in the variable "pathways" ) as input for the intersect argument:-

```R
upset(method_results_table, 
          intersect = pathways, 
          mode = "inclusive_intersection",
          intersections = list(pathways[1], pathways[2], c(pathways[1], pathways[2])))
```
I get the error, with an **empty graph**

```R
Warning messages:
1: In max(f) : no non-missing arguments to max; returning -Inf
2: Computation failed in `stat_summary()`:
argument must be coercible to non-negative integer 
```

Case 2:- I just give the two specific sets as input for 'intersect':-

```R
upset(method_results_table, 
          intersect = pathways[c(1,2)], 
          mode = "inclusive_intersection",
          intersections = list(pathways[1], pathways[2], c(pathways[1], pathways[2])))
```

it works perfectly!

![graph](https://user-images.githubusercontent.com/52196306/110203660-65ea1a80-7e95-11eb-846e-24bcc992d569.png)

ComplexUpset version: ‘1.1.0’

Some other information that might be useful

```R
> summary(method_results_table)
 R_HSA_9617828   R_HSA_6791312   R_HSA_9614085   R_HSA_3700989   R_HSA_69563     R_HSA_69580     R_HSA_69615    
 Mode :logical   Mode :logical   Mode :logical   Mode :logical   Mode :logical   Mode :logical   Mode :logical  
 FALSE:637       FALSE:606       FALSE:588       FALSE:291       FALSE:589       FALSE:589       FALSE:587      
 TRUE :17        TRUE :48        TRUE :66        TRUE :363       TRUE :65        TRUE :65        TRUE :67       
 R_HSA_69206     R_HSA_69242     R_HSA_2559583   GS_R_HSA_69242 
 Mode :logical   Mode :logical   Mode :logical   Mode :logical  
 FALSE:523       FALSE:493       FALSE:489       FALSE:493      
 TRUE :131       TRUE :161       TRUE :165       TRUE :161

> class(rownames(method_results_table))
[1] "character"
> class(method_results_table)
[1] "data.frame"
> dim(method_results_table)
[1] 654  11
```

**Comment by krassowski** [comment]
Please provide a minimal dataset that allows to reproduce this issue. Please read https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example.

**Comment by sharathkmenon** [comment]
```R
library(ComplexUpset)
test <- data.frame(matrix(data = TRUE, nrow=10, ncol=4))
temp <- colnames(test)
test[10,1] = FALSE
test[9,2] = FALSE
```

Case 1:- This gives ar error
```R
upset(test, temp, mode = "inclusive_intersection", intersections = list(temp[1], temp[2], c(temp[1], temp[2])))
```
```R
Warning messages:
1: In max(f) : no non-missing arguments to max; returning -Inf
2: Computation failed in `stat_summary()`:
argument must be coercible to non-negative integer 
```


Case 2:- This does not give an error

```R
upset(test, temp[c(1,2)], mode = "inclusive_intersection", intersections = list(temp[1], temp[2], c(temp[1], temp[2])))
```
![test](https://user-images.githubusercontent.com/52196306/110206098-d5fe9d80-7ea1-11eb-87ae-ae52215a36df.png)




Side note:-
In this dataset, sets "X3" and "X4" are exactly the same. If I try plotting the same for X3 and X4 (again, while just giving the two as input to 'intersect') only the intersection is displayed. 
```R
upset(test, temp[c(3,4)], mode = "inclusive_intersection", intersections = list(temp[3], temp[4], c(temp[3], temp[4])))
```

![test_2](https://user-images.githubusercontent.com/52196306/110206130-fe869780-7ea1-11eb-8ad3-d692d21d1ce9.png)

**Comment by krassowski** [comment]
Thank you, this is very helpful. Indeed the first case should work. Do I understand correctly that you would expect the first and second case to produce exactly the same plot?

**Comment by sharathkmenon** [comment]
Yes!
Other sets should not change the values as the mode is set to "inclusive intersection". 
More importantly, I do not know why case 1 throws up an error in the first place, so I was hoping you could give some insight. 

There is also the minor issue which I mentioned as a side note, where if the two sets are exactly the same, the plot only shows the intersection and does not display the number of elements in each set, despite the "intersections" argument. I added that because I thought they might have a common underlying cause

**Comment by krassowski** [comment]
So I had a simple fix coded locally that would make the behaviour consistent in this very example, conceptually by trimming the `intersect` argument value to only include sets that are included in the `intersections`. It looked like this:

```R
    if (specific_intersections) {
        sets_from_manual_intersections = unique(unlist(intersections))
        sets_from_intersect = unique(intersect)
        extra_sets = setdiff(sets_from_intersect, sets_from_manual_intersections)
        if (length(extra_sets) != 0) {

            warning(
                paste(
                    '`intersect` contains additional sets that were not used in `intersections`:',
                    paste(extra_sets, collapse=', ')
                )
            )
            intersect = intersect[intersect %in% sets_from_manual_intersections]
        }
    }
```

However, I realised that this is NOT the intended behaviour for two reasons:

1. There would be no difference between providing the extra sets in `intersect` and not doing so (except for the annoying warning); however, users might want to show the extra sets considered to provide more context.
2. This behaviour would be faulty when an additional set of cardinality were to be included

To illustrate point (2), let's consider:

```R
upset(
    test,
    temp,
    mode = "inclusive_intersection",
    intersections = list(
        temp[1],
        temp[2],
        c(temp[1], temp[2]),
        c(temp[1], temp[2], temp[3])
    )
)
```

which with the patch as mentioned above would result in:

![image](https://user-images.githubusercontent.com/5832902/113481934-36c6c900-9494-11eb-95e4-c3a594cf8bca.png)

However, user might be expecting to instead get:

![image](https://user-images.githubusercontent.com/5832902/113481963-5b22a580-9494-11eb-96d4-a3ba2ec3b043.png)

Therefore I went back to the drawing board. It turns out that the core of the problem is that providing the intersections manually means we have to internally switch to the same mechanisms as is being used when `intersections='all'` in contrast to the default `intersections='observed'`, because the (implicit) user expectation is that any intersection can be provided when specifying them manually. I have a solution worked out, but it needs some more polishing.
